internal/app/server/app.go

```diff
@@ -18,10 +21,23 @@ import (
 
 type App struct {
 	httpServer *http.Server
+	dbPool     *sql.DB
+	services   *service.Services
 }
 
-func NewApp() *App {
-	return &App{}
+func NewApp(cfg *config.Config) *App {
+	if cfg.DatabaseDSN == "" {
+		return &App{}
+	}
+	pool, err := connection.NewDBPool(cfg.DatabaseDSN)
+	if err != nil {
+		log.Fatalf("%s", err.Error())
```

Comment:
Никаких Fatal за пределами main()
```suggestion
		return err
```